test/TelegramPatternTests.hs

```diff
@@ -0,0 +1,114 @@
+module TelegramPatternTests where
+
+import Data.Functor.Identity (Identity)
+import qualified Data.Text.IO as TIO
+import Logger (Handle(..), Priority(Debug))
+import Telegram.Echo (echo)
+import Telegram.TelegramHandle (TelegramHandle(..))
+import Telegram.Types
+    ( TelegramToken(TelegramToken),
+      UpdateId(UpdateId),
+      HelpMessage(HelpMessage),
+      StatusResult(StatusResult) )
+import Test.Hspec (describe, hspec, it, shouldBe)
+import UsersLists
+    ( RepeatsNum(RepeatsNum), ChatId(ChatId), Repeats(Repeats) )
+import Telegram.Responses
+    ( TelegramUpdate(TelegramUpdate),
+      TelegramUser(TelegramUser, telegramUserId),
+      TelegramCallbackQuery(..),
+      TelegramMessage(TelegramMessage) )
+
+
+logHandle :: Handle Identity
+logHandle = Handle {priority = Debug, Logger.log = \prior message -> return ()}
+
+telegramHandle :: TelegramHandle Identity
+telegramHandle =
+    TelegramHandle
+        { getMe = \logHandle token -> return Nothing
+        , sendKeyboard = \logHandle token chatId -> return Nothing
+        , getUpdates = \logHandle token updateId -> return Nothing
+        , getLastUpdateId = \logHandle updates -> return Nothing
+        , sendTextMessage =
+              \logHandle token chatId text entities -> return Nothing
+        , sendAnimationMessage =
+              \logHandle token chatId animation caption -> return Nothing
+        , sendAudioMessage =
+              \logHandle token chatId audio caption -> return Nothing
+        , sendVideoMessage =
+              \logHandle token chatId video caption -> return Nothing
+        , sendDocumentMessage =
+              \logHandle token chatId document caption -> return Nothing
+        , sendPhotoMessage =
+              \logHandle token chatId photo caption -> return Nothing
+        , sendStickerMessage = \logHandle token chatId sticker -> return Nothing
+        , sendVideoNoteMessage =
+              \logHandle token chatId videoNote -> return Nothing
+        , sendVoiceMessage =
+              \logHandle token chatId voice caption -> return Nothing
+        , sendContactMessage = \logHandle token chatId contact -> return Nothing
+        , sendLocationMessage =
+              \logHandle token chatId location -> return Nothing
+        , sendVenueMessage = \logHandle token chatId venue -> return Nothing
+        }
+
+
+echoTelegramTests :: IO ()
+echoTelegramTests =
```

Comment:
не хватает теста когда в апдейте Nothing
![image](https://user-images.githubusercontent.com/11536292/134617550-14d4d926-4796-4fcf-8f6f-0f35f8cc838f.png)

Reply:
![image](https://user-images.githubusercontent.com/61455510/134625620-8e4e857c-a092-4815-90dc-90eb2fcca145.png)
это как раз есть. А вот тест, где в функция принимает Nothing вместо UpdateId, нет.

Reply:
А, поняла) круто что еще кейсов добавил. 
С названиями тестов еще немного беда, потому что они содержат конкретные значения. Лучше всегда называть по смыслу кейса, например "should not change UpdateId and repeatsList,  because don't have updates" вместо указания `(Nothing,[])`. Так по названию будет понятнее логика, почему результат именно такой.